### PR TITLE
KEP 10 - CLI enhancements to retrieve Keptn status information

### PR DESCRIPTION
**Pull request by checkelmann**
Hi,

I want to add a KEP for the CLI Enhancements we had discussed in slack a couple of weeks ago.

### CODE REVIEW

File: text/0010-keptn-cli-enhancements.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+## Motivation
+
+Right now when I'm using keptn within a CI/CD Pipeline I want to check if my service is already onboarded in keptn, if not it should be done automatically.
+From Ops-Perspective I want to get a list of all projects which are using keptn. Like in helm a keptn list command to retrieve a list of all projects and for more details a keptn get/details <project> command which returns all services within the project would be benificial.
+As workaround I'm checking if in my kubernetes cluster a lighthouse-configmap is existing, but this is not a pretty solution.
+
+## Explanation
+
+```bash
+# keptn list
```
Comment (christian-kreuzberger-dtx) [inline text/0010-keptn-cli-enhancements.md]: I'd like to be a little bit more generic here, e.g.:
```console
keptn get projects
```

File: text/0010-keptn-cli-enhancements.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Project             Onboarded                       Git Repository
+sockshop            2020-02-03 11:50                https://xyz/project.git
+```
+
+```bash
+# keptn get sockshop
```
Comment (christian-kreuzberger-dtx) [inline text/0010-keptn-cli-enhancements.md]: To be inline with my proposed syntax above:
```console
keptn get project sockshop
```

File: text/0010-keptn-cli-enhancements.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Project: sockshop
+Onboarded: 2020-02-03 11:50
+Git Repository: https://xyz/project.git
+Services:
+- carts-service
+- payment-service
+```
+
+```bash
+# keptn get sockshop service carts-service
```
Comment (christian-kreuzberger-dtx) [inline text/0010-keptn-cli-enhancements.md]: Same here:
```console
keptn get service carts-service --project=sockshop
```

**Comment by christian-kreuzberger-dtx** [comment]
Thanks, that's a good proposal. I've added some comments such that we are a little bit more generic with the CLI syntax. This would also make it feel a little bit more like `kubectl` (e.g., `kubectl get services`, `kubectl get deployments`, ...).

File: text/0010-keptn-cli-enhancements.md (hunk 1 above)
Comment (johannes-b) [reply text/0010-keptn-cli-enhancements.md]: Like with `kubectl` no differentiation between singular and plural should be made. 
For example, `keptn get projects` returns the same like `keptn get project`

**Comment by checkelmann** [comment]
> Thanks, that's a good proposal. I've added some comments such that we are a little bit more generic with the CLI syntax. This would also make it feel a little bit more like `kubectl` (e.g., `kubectl get services`, `kubectl get deployments`, ...).

Hey Christian,

for me as "customer", the actual syntax does not matter as long as it is getting implemented ;). But I agree, a kubectl style would be good.

File: text/0010-keptn-cli-enhancements.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+Project: sockshop
+Service: carts-service
+Last artifact: myimage:v1.0.2
```
Comment (christian-kreuzberger-dtx) [inline text/0010-keptn-cli-enhancements.md]: Details like

* Last artifact
* last deployment
* etc...

should be listed per stage

**Pull request merged**